charts/connectors/templates/deployment.yaml

```diff
@@ -0,0 +1,208 @@
+            timeoutSeconds: {{ .Values.deployment.livenessProbe.timeoutSeconds }}
+          readinessProbe:
+            httpGet:
+              path: /
+              port: rest-api
+              scheme: HTTP
+            initialDelaySeconds: {{ .Values.deployment.readinessProbe.initialDelaySeconds }}
+            failureThreshold: {{ .Values.deployment.readinessProbe.failureThreshold }}
+            periodSeconds: {{ .Values.deployment.readinessProbe.periodSeconds }}
+            successThreshold: {{ .Values.deployment.readinessProbe.successThreshold }}
+            timeoutSeconds: {{ .Values.deployment.readinessProbe.timeoutSeconds }}
+          ports:
+          {{- range $port := .Values.service.ports }}
+            - containerPort: {{ $port.port }}
+              name: {{ $port.name }}
+              protocol: TCP
+          {{- end }}
+          resources:
+            requests: {{ toYaml .Values.container.resources.request | nindent 14 }}
+            limits: {{ toYaml .Values.container.resources.limits | nindent 14 }}
+          terminationMessagePath: /dev/termination-log
+          terminationMessagePolicy: File
+          volumeMounts:
+          {{- if and .Values.auth.sasl.userName .Values.auth.sasl.mechanism .Values.auth.sasl.secretRef }}
+            - mountPath: /opt/kafka/connect-password/rc-credentials
+              name: rc-credentials
+          {{- end }}
+            - mountPath: /tmp
+              name: rp-connect-tmp
+      dnsPolicy: ClusterFirst
+      restartPolicy: {{ .Values.deployment.restartPolicy }}
+    {{- with .Values.deployment.schedulerName }}
+      schedulerName: {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.deployment.nodeSelector }}
+      nodeSelector: {{- . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets: {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.deployment.securityContext }}
+      securityContext: {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.deployment.tolerations }}
+      tolerations: {{- toYaml . | nindent 8 }}
+    {{- end }}
+      topologySpreadConstraints:
+    {{- range $spread := .Values.deployment.topologySpreadConstraints }}
+        - labelSelector:
+            matchLabels:  {{ include "connectors-pod-labels" $root | nindent 14 }}
+          maxSkew: {{ $spread.maxSkew }}
+          topologyKey: {{ $spread.topologyKey }}
+          whenUnsatisfiable: {{ $spread.whenUnsatisfiable }}
+    {{- end }}
+      volumes:
+      {{- if and .Values.auth.sasl.userName .Values.auth.sasl.mechanism .Values.auth.sasl.secretRef }}
+        - name: rc-credentials
+          secret:
+            defaultMode: 292
+            secretName: {{ .Values.auth.sasl.secretRef }}
+      {{- end }}
+        - emptyDir:
+            medium: Memory
+            sizeLimit: {{ .Values.deployment.connectTmpSizeLimit }}
+          name: rp-connect-tmp
```

Comment:
Should this be templated? I can imagine users that don't want ram disks when they have nvme disks.

Reply:
ram disks is quicker than nvme disks. I can templated it sure.